pilot/pkg/security/authz/model/permission_test.go

```diff
@@ -760,6 +761,84 @@ func TestPermission_Generate(t *testing.T) {
                   exactMatch: /world
                   name: :path`,
 		},
+		{
+			name: "permission with forDeny",
+			permission: &Permission{
+				Methods: []string{"GET"},
+				Constraints: []KeyValues{
+					{
+						attrDestIP: Values{
+							Values: []string{"1.2.3.4", "5.6.7.8"},
+						},
+					},
+				},
+			},
+			forDeny: true,
+			wantYAML: `
+        andRules:
+          rules:
+          - orRules:
+              rules:
+              - header:
+                  exactMatch: GET
+                  name: :method
+          - orRules:
+              rules:
+              - destinationIp:
+                  addressPrefix: 1.2.3.4
+                  prefixLen: 32
+              - destinationIp:
+                  addressPrefix: 5.6.7.8
+                  prefixLen: 32`,
+		},
+		{
```

Comment:
Would be nice to add a simple permission like above (even better, change the rule above to use HTTP attributes and use the same permission for the test case with `forTCPFilter=true`) to see the key different, before get into a complicated one like the next.

Reply:
Updated the test, also this is more thoroughly tested in pilot/pkg/security/authz/builder/testdata/v1beta1/action-deny-HTTP-for-TCP-filter-in.yaml